ntlm3/ntlm.py

```diff
@@ -134,282 +56,252 @@
+    :param kwargs: An optional dictionary (with default values) which can be used to configure
+
+                    ntlm_compatibility      - The Lan Manager Compatibility Level to use withe the auth message - Default 3
+                                                This is set by an Administrator in the registry key
+                                                'HKLM\SYSTEM\CurrentControlSet\Control\Lsa\LmCompatibilityLevel'
+                                                The values correspond to the following;
+                                                0 : Clients use LM and NTLM authentication, but they never use NTLMv2 session security.
+                                                    Domain controllers accept LM, NTLM, and NTLMv2 authentication.
+                                                1 : Clients use LM and NTLM authentication, and they use NTLMv2 session security if the
+                                                    server supports it. Domain controllers accept LM, NTLM, and NTLMv2 authentication.
+                                                2 : Clients use only NTLM authentication, and they use NTLMv2 (NTLM2 in this code) session
+                                                    security if the server supports it. Domain controller accepts LM, NTLM, and NTLMv2 authentication.
+                                                3 : Clients use only NTLMv2 authentication, and they use NTLMv2 session security if the server
+                                                    supports it. Domain controllers accept LM, NTLM, and NTLMv2 authentication.
+                                                4 : Clients use only NTLMv2 authentication, and they use NTLMv2 session security if the server
+                                                    supports it. Domain controller refuses LM authentication responses, but it accepts NTLM and NTLMv2
+                                                5 : Clients use only NTLMv2 authentication, and they use NTLMv2 session security if the server
+                                                    supports it. Domain controller refuses LM and NTLM authentication responses, but it accepts NTLMv2.
+
+                    server_certificate_hash - The SHA256 hash string of the server certificate (DER encoded) NTLM is authenticating to. This is used to add
+                                              to the gss_channel_bindings_struct for Channel Binding Tokens support. If none is passed through then python-ntlm3
+                                              will not use Channel Binding Tokens when authenticating with the server which could cause issues if it is set to
+                                              only authenticate when these are present. This is only used for NTLMv2 authentication.
+"""
+def create_NTLM_AUTHENTICATE_MESSAGE(server_challenge, user, domain, password, server_negotiate_flags, server_target_info=None, **kwargs):

```

Comment:
I don't picture this causing issues but this method now takes in more arguments that are optional, hopefully not too bad for people who want to use the newer version.